src/Angor/Client/Pages/Spend.razor

```diff
@@ -781,12 +801,49 @@
                 {
                     item.IsSpent = true;
 
-                    updateUnconfirmedOutbound |= LinqExtension.Remove(unconfirmedOutbound, new Outpoint(item.Trxid, item.Outputindex));
+                    updateUnconfirmedOutbound |= unconfirmedOutbound.TryRemoveOutpoint(new Outpoint(item.Trxid, item.Outputindex));
+
+                    // try to resolve the destination
+                    var spentInTransaction = await _IndexerService.GetTransactionInfoByIdAsync(output.SpentInTransaction);
+
+                    var input = spentInTransaction?.Inputs.FirstOrDefault(input => input.InputTransactionId == item.Trxid && input.InputIndex == item.Outputindex);
+
+                    var investmentTransaction = trxsHexCache.TryGet(item.Trxid);
+                    if (investmentTransaction == null)
+                    {
+                        investmentTransaction = network.CreateTransaction(await _IndexerService.GetTransactionHexByIdAsync(item.Trxid));
```

Comment:
please move await _IndexerService.GetTransactionHexByIdAsync(item.Trxid) to a different line